signal_id/mask.py

```diff
@@ -115,7 +115,7 @@ def __init__(self, data, thresh=None, backup=True, *args):
         if isinstance(data, SpectralCube):
                 self.from_spec_cube(data, thresh=thresh, *args)
 
-        elif isinstance(data, str):
+        elif isinstance(data, str) or isinstance(data, unicode):
```

Comment:
we should be using `isinstance(data, six.string_types)` here (`from astropy.extern import six`, I think)